lib/request-etag.js

```diff
@@ -40,6 +40,7 @@ function get(cache) {
 
     var cacheHit = cache.get(options);
     if (cacheHit) {
+      options.headers = options.headers || {};

```

Comment:
I think that this cannot be done inside this `if` block otherwise we may end up in a bad situation when the response of a headerless cached request changes.
The problem is that, in this situation, we will store the value in the cache with a key (with headers) different from the key that we use (without headers) when querying the cache.
Not easy to explain here. Let me know if this makes sense or not..